### PR TITLE
pgfkeys: fix two handler problems

### LINKED ISSUE #654: /.code keys don't respect /.initial
If a keys has been defined with `/.code`, the initial value can't be set with `/.initial`.  The following MWE shoud throw an undefined control sequence error when trying to set `/.initial` but that does not happen.
```latex
\input pgfkeys.tex
\pgfkeys{foo/.code=\ERROR}
\pgfkeys{foo/.initial=bar}
%\pgfkeys{foo=bar} % BOOM
\bye
```
This behaviour should either be fixed or documented.

### LINKED ISSUE #912: Document brace stripping in pgfkeys
Version: 3.1.5b 

`pgfkeys` will strip braces around single arguments, e.g.
```latex
\pgfset{
  /test/foo/.code=bar,
  /test/.style n args={1}{/test/.cd,#1},
  /test={foo},
}
```
will result in
```
! Package pgfkeys Error: I do not know the key '/test/f' and I am going to igno
re it. Perhaps you misspelled it.
```
This is totally expected when you know that `/.style` handlers are implemented in terms of `/.code` handlers and usual TeX argument brace stripping applies. However, this is not mentioned clearly in the manual.

### PR DESCRIPTION
**Pull request by muzimuzhi**
See commit messages and comments under corresponding issues for more info.

Closes #654
Closes #912

### CODE REVIEW

**Review by hmenke — CHANGES_REQUESTED** [review]
Please revert the whitespace-only changes.  I don't care about “consistency”, I want a minimal diff.

**Comment by muzimuzhi** [comment]
> Please revert the whitespace-only changes. I don't care about “consistency”, I want a minimal diff.

That's not for consistency, but for readability. I can live with in-consistency in the code base and I will not add as many as whitespace-only changes just for “consistency”.

But from my point of view, https://github.com/pgf-tikz/pgf/pull/913/commits/1b420435c47883620ede1156e56005275d73561f do makes reading and understanding implementation of `.add code` easier.

File: tex/generic/pgf/utilities/pgfkeys.code.tex [hunk 1]
```diff
@@ -738,30 +742,30 @@
...
-  }{%
-    % support for \pgfkeysdefargs:
+      % support for \pgfkeysndefargs:
+      \pgfkeyslet{\pgfkeyscurrentpath/.@@body}{\pgfkeys@global@temp}%
+    }{%
+      % support for \pgfkeysdefargs:
       \pgfkeyslet{\pgfkeyscurrentpath/.@cmd}{\pgfkeys@global@temp}%
-  }%
+    }%
   }%
```
Comment (muzimuzhi) [inline tex/generic/pgf/utilities/pgfkeys.code.tex:760]: For example, before that commit, it's hard to identify which `}` ends the second argument of `\pgfkeysifdefined`.

**Comment by hmenke** [comment]
Merged in https://github.com/pgf-tikz/pgf/commit/ab25066f631cd8137a4b7c91aef4b0ccfc97f991

**Comment by muzimuzhi** [comment]
I respect your decision, though ... .

Avoiding whitespace-only changes like tab-to-space replacement is reasonable, but avoiding every such change is not always good, I'm afraid. For example, when the definition is long and complex, changing
```tex
\def\cmd{%
<definition>
}
```
to
```tex
\def\cmd{%
<definition>
}
```
helps everyone who reads the code in the future. Utilities are for users, while code is written and maintained for people to read.

I'm not expecting to influence your decision, but restate my idea with an example. If anything looks offending, that's all my fault (maybe partly the fault of my influent hence sometimes unconsciously impolite English).

**Comment by muzimuzhi** [comment]
I was convinced. unnecessary changes made the tracing work harder. For large repo with long history and perhaps in-sufficient maintenance support, please the one who traces problems is more important than improve code consistency and/or readability. For example, the change history of [`\pgfpointintersectionoflines`](https://github.com/pgf-tikz/pgf/blame/f5ee9149477346a61949b6f1bc81757a7ebc9da8/tex/generic/pgf/basiclayer/pgfcorepoints.code.tex#L197).

**Pull request closed without merging**